internal/webhook/management_webhook.go

```diff
@@ -53,13 +56,91 @@ func (*ManagementValidator) ValidateCreate(_ context.Context, _ runtime.Object)
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
-func (*ManagementValidator) ValidateUpdate(_ context.Context, _ runtime.Object, _ runtime.Object) (admission.Warnings, error) {
+func (v *ManagementValidator) ValidateUpdate(ctx context.Context, _, newObj runtime.Object) (admission.Warnings, error) {
+	const invalidMgmtMsg = "the Management is invalid"
+
+	mgmt, ok := newObj.(*hmcv1alpha1.Management)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected Management but got a %T", newObj))
+	}
+
+	if mgmt.Status.Release == "" {
+		return nil, nil // nothing to do at the moment
+	}
+
+	skipValidation := true
+	capiTplName := ""
+	for cn, c := range mgmt.Status.Components {
+		if cn == hmcv1alpha1.CoreCAPIName && c.Success {
+			skipValidation = false
+			capiTplName = c.Template
+			break
+		}
+	}
+
+	if skipValidation {
+		return nil, nil // either no status, or capi has been failed to deploy
+	}
+
+	ptpls := new(hmcv1alpha1.ProviderTemplateList)
+	if err := v.List(ctx, ptpls); err != nil {
+		return nil, fmt.Errorf("failed to list ProviderTemplates: %v", err)
+	}
+
+	if len(ptpls.Items) == 0 {
+		return nil, nil // nothing to do
+	}
+
+	name2Tpl := make(map[string]*hmcv1alpha1.ProviderTemplate, len(ptpls.Items)-1)
+	capiRequiredVersion := new(semver.Version)
+	for _, v := range ptpls.Items { // cluster-scoped
+		if v.Name != capiTplName {
+			name2Tpl[v.Name] = &v
+			continue
+		}
+
+		if v.Status.CAPIVersion == "" {
+			return nil, nil // nothing to validate against
+		}
+
+		var err error
+		capiRequiredVersion, err = semver.NewVersion(v.Status.CAPIVersion)
+		if err != nil { // should never happen
+			return nil, fmt.Errorf("%s: invalid CAPI version %s in the ProviderTemplate %s to be validated against: %v", invalidMgmtMsg, v.Status.CAPIVersion, v.Name, err)
+		}
+	}
+
+	var wrongVersions error
+	for _, c := range mgmt.Status.Components {
```

Comment:
what do you intent to validate here? core capi update with the existing mgmt providers?